routes/api.php

```diff
@@ -15,16 +15,53 @@
 |
 */
 
+
 Route::middleware('auth:api')->get('/user', function (Request $request) {
     return $request->user();
 });
 
-Route::get('/recordings', [RecordingsController::class, 'getAllRecordings']);
+Route::group([
+    'middleware' => 'auth',
+], function ($router)
+{
+    Route::get('/recordings', [RecordingsController::class, 'getAllRecordings']);
+
+    Route::get('/recordings/artist/{name}', [RecordingsController::class, 'getRecordingByArtist']);
+
+    Route::post('/recordings', [RecordingsController::class, 'createRecording']);
+
+    Route::put('/recordings/{recording}', [RecordingsController::class, 'updateRecording']);
+}
+);
 
+Route::group([
```

Comment:
Please, move entire group to web.php file and remove api middleware, since these routes are not part of the api.